src/main/java/org/opentripplanner/middleware/triptracker/interactions/busnotifiers/UsRideGwinnettNotifyBusOperator.java

```diff
@@ -0,0 +1,210 @@
+import org.opentripplanner.middleware.utils.JsonUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.net.URI;
+import java.time.Duration;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Map;
+
+import static org.opentripplanner.middleware.utils.ConfigUtils.getConfigPropertyAsText;
+import static org.opentripplanner.middleware.utils.ItineraryUtils.getRouteIdFromLeg;
+import static org.opentripplanner.middleware.utils.ItineraryUtils.isBusLeg;
+
+/**
+ * If conditions are correct, notify a bus operator of a traveler waiting to board at a given stop.
+ */
+public class UsRideGwinnettNotifyBusOperator implements BusOperatorInteraction {
+
+    public static boolean IS_TEST = false;
+
+    private static final Logger LOG = LoggerFactory.getLogger(UsRideGwinnettNotifyBusOperator.class);
+
+    private static final String US_RIDE_GWINNETT_BUS_OPERATOR_NOTIFIER_API_URL
+        = getConfigPropertyAsText("US_RIDE_GWINNETT_BUS_OPERATOR_NOTIFIER_API_URL", "not-provided");
+
+    private static final String US_RIDE_GWINNETT_BUS_OPERATOR_NOTIFIER_API_KEY
+        = getConfigPropertyAsText("US_RIDE_GWINNETT_BUS_OPERATOR_NOTIFIER_API_KEY", "not-provided");
+
+    public static List<String> US_RIDE_GWINNETT_QUALIFYING_BUS_NOTIFIER_ROUTES = getBusOperatorNotifierQualifyingRoutes();
+
+    public static final int ACCEPTABLE_AHEAD_OF_SCHEDULE_IN_MINUTES = 15;
+
+    /**
+     * Headers that are required for each request.
+     */
+    private static final Map<String, String> BUS_OPERATOR_NOTIFIER_API_HEADERS = Map.of(
+        "Ocp-Apim-Subscription-Key", US_RIDE_GWINNETT_BUS_OPERATOR_NOTIFIER_API_KEY,
+        "Content-Type", "application/json"
+    );
+
+
+    /**
+     * Get the routes which qualify for bus operator notifying from the configuration. The configuration value is
+     * expected to be a comma separate list of agency id (as provided by OTP not agency) and route id e.g.
+     * <p>
+     * GwinnettCountyTransit:360,GwinnettCountyTransit:40,GwinnettCountyTransit:25
+     */
+    private static List<String> getBusOperatorNotifierQualifyingRoutes() {
+        String busOperatorNotifierQualifyingRoutes
+            = getConfigPropertyAsText("US_RIDE_GWINNETT_BUS_OPERATOR_NOTIFIER_QUALIFYING_ROUTES");
+        if (busOperatorNotifierQualifyingRoutes != null) {
+            return Arrays.asList(busOperatorNotifierQualifyingRoutes.split(","));
+        }
+        return new ArrayList<>();
+    }
+
+    /**
+     * Stage notification to bus operator by making sure all required conditions are met.
+     */
+    public void sendNotification(TripStatus tripStatus, TravelerPosition travelerPosition) {
+        var routeId = getRouteIdFromLeg(travelerPosition.nextLeg);
+        try {
+            if (
+                isBusLeg(travelerPosition.nextLeg) &&
+                isWithinOperationalNotifyWindow(tripStatus, travelerPosition) &&
```

Comment:
I have to give this one a thought. This condition seems more restrictive geographically than `isApproachingEndOfLeg(travelerPosition)` that must be met before executing this code. But then, to compute whether someone is on-time or ahead of schedule, we'd have to use the real-time bus arrival time vs. the current time. What are your thoughts on that, @br648?

Reply:
@binh-dam-ibigroup In the interest of time, if you would like to provide psuedo code/logic I could implement something for review as part of our meeting tomorrow?

Also, is my assumption above about real-time correct?

Reply:
@binh-dam-ibigroup Refactor in: https://github.com/ibi-group/otp-middleware/pull/233/commits/56e6c1553f1c9d0af609ad45a2890aff2d341aaf